lib/src/insertion_order_parser.dart

```diff
@@ -0,0 +1,164 @@
+
+    for (Map segmentMap in map['budgetSegments'] ?? []) {
+      budget.budgetSegments.add(_createBudgetSegment(segmentMap));
+    }
+    return budget;
+  }
+
+  static InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment
+      _createBudgetSegment(Map<String, dynamic> map) {
+    if (map == null || map.isEmpty) {
+      return InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment();
+    }
+
+    final budgetSegment =
+        InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment()
+          ..budgetAmountMicros = map['budgetAmountMicros'] ?? _emptyEntry
+          ..description = map['description'] ?? _emptyEntry
+          ..campaignBudgetId = map['campaignBudgetId'] ?? _emptyEntry
+          ..dateRange = _createDateRange(map['dateRange']);
+
+    return budgetSegment;
+  }
+
+  static InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange
+      _createDateRange(Map<String, dynamic> map) {
+    if (map == null || map.isEmpty) {
+      return InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange();
+    }
+
+    final dateRange =
+        InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange()
+          ..startDate = _createDate(map['startDate'])
+          ..endDate = _createDate(map['endDate']);
+
+    return dateRange;
+  }
+
+  static InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange_Date
+      _createDate(Map<String, dynamic> map) {
+    if (map == null || map.isEmpty) {
+      return InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange_Date();
+    }
+
+    final date =
+        InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange_Date()
+          ..year = map['year'] ?? _emptyEntry
+          ..month = map['month'] ?? _emptyEntry
+          ..day = map['day'] ?? _emptyEntry;
+
+    return date;
+  }
+
+  static InsertionOrder_EntityStatus _createEntityStatus(String target) {
+    return _entityStatusMap[target] ?? InsertionOrder_EntityStatus.values[0];
+  }
+
+  static InsertionOrder_Pacing_PacingPeriod _createPacingPeriod(String target) {
+    return _pacingPeriodMap[target] ??
+        InsertionOrder_Pacing_PacingPeriod.values[0];
+  }
+
+  static InsertionOrder_Pacing_PacingType _createPacingType(String target) {
+    return _pacingTypeMap[target] ?? InsertionOrder_Pacing_PacingType.values[0];
+  }
+
+  static InsertionOrder_InsertionOrderBudget_BudgetUnit _createBudgetUnit(
+      String target) {
+    return _budgetUnitMap[target] ??
+        InsertionOrder_InsertionOrderBudget_BudgetUnit.values[0];
+  }
+
+  static InsertionOrder_InsertionOrderBudget_InsertionOrderAutomationType
+      _createAutomationType(String target) {
+    /// Returns default value if [target] is null
+    if (target == null) {
+      return InsertionOrder_InsertionOrderBudget_InsertionOrderAutomationType
+          .values[0];
```

Comment:
Default should be INSERTION_ORDER_AUTOMATION_TYPE_NONE

Reply:
Yes, automationType.values[0] here evaluates to INSERTION_ORDER_AUTOMATION_TYPE_NONE